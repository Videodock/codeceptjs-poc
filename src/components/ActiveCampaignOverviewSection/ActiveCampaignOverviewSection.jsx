import React from 'react';
import { makeStyles } from '@material-ui/styles';
import { useTranslation } from 'react-i18next';
import * as PropTypes from 'prop-types';
import { Typography } from '@material-ui/core';
import Slider from '../Slider';
import styles from './ActiveCampaignOverviewSection.styles';

const useStyles = makeStyles(styles);

export default function ActiveCampaignOverviewSection ({ children, campaignsToShow = 4 }) {
  const { t }   = useTranslation('campaign');
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Typography className={classes.heading} variant="h2">
        {t('overview.heading')}
      </Typography>
      <Slider slidesToShow={campaignsToShow} framePadding="20px 25px 0 25px">
        {children}
      </Slider>
    </div>
  );
}

ActiveCampaignOverviewSection.propTypes = {
  campaignsToShow: PropTypes.number,
};
