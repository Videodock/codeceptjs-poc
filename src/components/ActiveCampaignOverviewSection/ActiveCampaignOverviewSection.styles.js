import { createStyles } from '@material-ui/styles';

export default ({ palette }) => createStyles({
  '@global': {
    ".slider-control-bottomcenter ul": {
      top: '60px !important',
    },
  },
  root: {
    backgroundColor: palette.common.faintGrey,
    padding        : 96,
    paddingTop     : 48,
  },
  heading: {
    marginLeft: 25,
  },
});
