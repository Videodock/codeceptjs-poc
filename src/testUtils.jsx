import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/styles';
import { I18nextProvider } from 'react-i18next';
import { ApolloProvider } from '@apollo/client';
import { BrowserRouter as Router } from 'react-router-dom';
import 'url-search-params-polyfill';
import i18n from '../src/i18n';
import client from './graphql';
import theme from './theme';
import store from './redux';

const wrapper = ({ children }) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>
          <Router>
            {children}
          </Router>
        </I18nextProvider>
      </ThemeProvider>
    </Provider>
  </ApolloProvider>
);

const customRender = (ui, options) => render(ui, { wrapper, ...options });

export { customRender as render };
