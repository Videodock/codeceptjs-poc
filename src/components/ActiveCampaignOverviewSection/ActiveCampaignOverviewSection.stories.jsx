import React from 'react';
import { storiesOf } from '@storybook/react';
import ActiveCampaignOverviewSection from './ActiveCampaignOverviewSection';

storiesOf('ActiveCampaignOverviewSection', module)
  .add('basic', () => <ActiveCampaignOverviewSection />);
