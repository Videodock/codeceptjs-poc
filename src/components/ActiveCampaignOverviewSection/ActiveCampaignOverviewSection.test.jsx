import React from 'react';
import { render } from '../../testUtils';
import ActiveCampaignOverviewSection from './ActiveCampaignOverviewSection';

describe('<ActiveCampaignOverviewSection />', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<ActiveCampaignOverviewSection />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
