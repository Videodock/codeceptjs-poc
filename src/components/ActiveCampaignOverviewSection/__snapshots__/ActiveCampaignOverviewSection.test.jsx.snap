// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActiveCampaignOverviewSection /> renders and matches snapshot 1`] = `
<div
  class="makeStyles-root-1"
>
  <h2
    class="MuiTypography-root makeStyles-heading-2 MuiTypography-h2"
  >
    Active campaigns
  </h2>
  <div
    class="slider"
    style="box-sizing: border-box; display: block; height: inherit; position: relative; width: 100%;"
    tabindex="0"
  >
    <div
      aria-atomic="true"
      aria-live="polite"
      style="position: absolute; left: -10000px; width: 1px; height: 1px; overflow: hidden;"
      tabindex="-1"
    >
      Slide 1 of 0
    </div>
    <div
      class="slider-frame"
      style="box-sizing: border-box; display: block; height: 100%; margin: 20px 25px 0px 25px; overflow: hidden; padding: 0px; position: relative; transform: translate3d(0, 0, 0);"
    >
      <ul
        class="slider-list"
        style="box-sizing: border-box; cursor: pointer; display: block; height: 0px; margin: 0px -8px; padding: 0px; position: relative; transform: translate3d(0px, 0px, 0); width: auto;"
      />
    </div>
    <div
      class="slider-control-centerright"
      style="position: absolute; top: 50%; right: 0px; transform: translateY(-50%);"
    >
      <button
        aria-label="Slide right"
        class="MuiButtonBase-root MuiIconButton-root makeStyles-slideButton-33"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="slider-control-bottomcenter"
      style="position: absolute; bottom: 0px; left: 50%; transform: translateX(-50%);"
    >
      <ul
        style="position: relative; top: -10px; display: flex; margin: 0px; padding: 0px; list-style-type: none;"
      >
        <li
          class="paging-item active"
        >
          <button
            aria-label="slide 1 bullet"
            style="cursor: pointer; opacity: 1; background: transparent; fill: black;"
            type="button"
          >
            <svg
              class="paging-dot"
              height="6"
              width="6"
            >
              <circle
                cx="3"
                cy="3"
                r="3"
              />
            </svg>
          </button>
        </li>
      </ul>
    </div>
    <style
      type="text/css"
    >
      .slider-slide &gt; img { width: 100%; display: block; }
          .slider-slide &gt; img:focus { margin: auto; }
    </style>
  </div>
</div>
`;
